### PR TITLE
feat: add key storage service

### LINKED ISSUE #339: Support backup for mnemonic
Backup root key

### LINKED ISSUE #342: Service for keys generation, signing
- [x] Generate mnemonic
- [x] Generate root key
- [x] Save encrypted root key to local storage
- [x] Create key management service and message signing

### PR DESCRIPTION
**Pull request by 0xmad**
## Explanation

This PR adds support for bip39 mnemonic phrase generation and key storage service.
This is a part of #333.

Details are below:
- [x] Add bip39 and support mnemonic phrase generation
- [x] Add service for key storage to generate key pair and sign messages
- [x] Implement backupable for key storage service  

## More Information

Closes #339 
Closes #342 

## Screenshots/Screencaps

N/A

## Manual Testing Steps

N/A

## Pre-Merge Checklist

- [x] PR template is filled out
- [ ] **IF** this PR fixes a bug, a test that _would have_ caught the bug has been added
- [x] PR is linked to the appropriate GitHub issue
- [x] PR has been added to the appropriate release Milestone

> PR template source from [github.com/MetaMask](https://github.com/MetaMask)

### CODE REVIEW

File: src/background/services/mnemonic/index.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import * as bip39 from "bip39";
+
+export const generateMnemonic = (): string => bip39.generateMnemonic();
+
+export const validateMnemonic = (mnemonic: string): boolean => bip39.validateMnemonic(mnemonic);
+
+export const mnemonicToSeed = (mnemonic: string): Promise<string> =>
+  bip39.mnemonicToSeed(mnemonic).then((bytes) => bytes.toString("hex"));
```
Comment (0xisk) [inline src/background/services/mnemonic/index.ts:8]:
```suggestion
import { generateMnemonic, validateMnemonic, mnemonicToSeed } from "bip39";

export const bip39GenerateMnemonic = (): string => generateMnemonic();

export const bip39ValidateMnemonic = (mnemonic: string): boolean => validateMnemonic(mnemonic);

export const bip39MnemonicToSeed = (mnemonic: string): Promise<string> =>
  mnemonicToSeed(mnemonic).then((bytes) => bytes.toString("hex"));
```

Comment (0xisk) [inline src/background/services/mnemonic/index.ts:8]: Do you think we should also enforce using {} braces here as well in functions as we done in conditions?

File: src/background/services/key/types.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export interface KeyPair {
+  publicKey: string;
+  secretKey: string;
```
Comment (0xisk) [inline src/background/services/key/types.ts:3]: Should we rename this to `privateKey`?

**Review by 0xisk — CHANGES_REQUESTED** [review]
Thanks @0xmad, I left some comments.

File: src/background/services/mnemonic/index.ts (hunk 1 above)
Comment (0xmad) [reply src/background/services/mnemonic/index.ts:8]: For arrow functions no. If there is more than 1 line we can use `function`.

File: src/background/services/key/types.ts (hunk 2 above)
Comment (0xmad) [reply src/background/services/key/types.ts:3]: The same fields are in tweetnacl library.

File: src/background/services/mnemonic/index.ts (hunk 1 above)
Comment (0xmad) [reply src/background/services/mnemonic/index.ts:8]: I don't think that this solution works. If we have another version of mnemonic generation, I'd keep generic names without implementation details.

Comment (0xisk) [reply src/background/services/mnemonic/index.ts:8]: Resolved because webpack will resolve this by tree-shaking.

**Review by 0xisk — APPROVED** [review]
Thanks @0xmad

**Pull request merged**